### DISCUSSION TITLE
What's the best way to combine 2 patterns to create a new one?

### DISCUSSION THREAD

**Discussion by itsjavi** [chakra-ui/panda#967] [Q&A]
is there any mechanism to combine pattern  e.g. "A" and a pattern "B" to create a derivate pattern?

I am thinking about something like: 

```
combinePatterns(...patterns: Partial<PatternConfig>[]) => PatternConfig
```

It would be like a pipeline of patterns that will apply transforms sequentially

**Comment by anubra266** [comment]
@itsjavi I'm not sure what you mean exactly. But you can extend patterns
https://panda-css.com/docs/customization/patterns#customizing-built-in-patterns

**Reply by itsjavi** [reply]
I was thinking about composability, with something like a pipeline of patterns: they will be applied sequentially and the function would return a combined version of them.

Let's say you have a "box" recipe and a "fancyBorders" recipe and you want to combine them to create "fancyBox" with the properties of both.

I had a more specific use case in mind last month, but I forgot :D. Nevermind, I think I can achieve that with creating a function like `combinePatterns`, but it might be too much complexity to add it to pandacss lib, or what do you think?

PS: The link you provided is only about extending built-in patterns, but not custom ones.

**Reply by heycassidy** [reply]
I would also find this useful. I've got a `focusOutline` pattern which generates a custom focus ring. I've also got a `textLink` which generates some styles for text links, like colors for different states, text-decoration, etc. I'd like to include the `focusOutline` pattern inside the `textLink` pattern, but haven't found a way to do that yet.
